mpas_analysis/ocean/ocean_modelvsobs.py

```diff
@@ -0,0 +1,274 @@
+    monthly_clim = ds_tslice.groupby('Time.month').mean('Time')
+    
+    dLongitude = 1.   # Target longitude spacing
+    dLatitude = 1.   # Target latitude spacing
+    
+    dtr = 180/np.pi
+    d2, inds2, lonTarg, latTarg = init_tree(lonCell*dtr,latCell*dtr, dLongitude, dLatitude)
+    d, inds, lonTargD, latTargD = init_tree(lonDATA, latDATA, dLongitude, dLatitude)
+    
+    nLon = lonTarg.shape[0]
+    nLat = lonTarg.shape[1]
+    SST_MODEL = np.zeros((len(outputTimes),nLon,nLat))
+    SST_DATA = np.zeros((len(outputTimes),nLon,nLat))
+    SST_BIAS = np.zeros((len(outputTimes),nLon,nLat))
+
+    for i, timestring in enumerate(outputTimes):
+        
+        monthsvalue = monthdictionary[timestring]
+        
+        if isinstance(monthsvalue, int):
+            modeldata = monthly_clim.sel(month=monthsvalue).time_avg_dThreshMLD.values
+            obsdata = monthly_clim_data.sel(month=monthsvalue).mld_dt_mean.values
+        
+        else:
+            modeldata = np.sum(dinmonth[monthsvalue-1]*monthly_clim.sel(month=monthsvalue).time_avg_activeTracers_temperature.values.T,axis=1) / np.sum(dinmonth[monthsvalue-1])
+            obsdata = np.nansum(daysarray[monthsvalue-1,:,:]*monthly_clim_data.sel(month=monthsvalue).SST.values,axis=0) / np.nansum(daysarray[monthsvalue-1,:,:],axis=0)
+        
+        SST_MODEL[i,:,:] = interp_fields(modeldata, d2, inds2, lonTarg)
+        SST_DATA[i,:,:] = interp_fields(obsdata.T.flatten(), d, inds, lonTargD)
+        
+    for i in range(len(outputTimes)):
+        SST_BIAS[i,:,:] = SST_MODEL[i,:,:] - SST_DATA[i,:,:]
+
+    clevsModelObs = config.getlist('sst_modelvsobs','clevsModelObs',listType=float)
+    cmap = plt.get_cmap(config.get('sst_modelvsobs','cmapModelObs'))
+    cmapIndices = config.getlist('sst_modelvsobs','cmapIndicesModelObs',listType=int)
+    cmapModelObs = cols.ListedColormap(cmap(cmapIndices),"cmapModelObs")
+
+    clevsDiff = config.getlist('sst_modelvsobs','clevsDiff',listType=float)
+    cmap = plt.get_cmap(config.get('sst_modelvsobs','cmapDiff'))
+    cmapIndices = config.getlist('sst_modelvsobs','cmapIndicesDiff',listType=int)
+    cmapDiff = cols.ListedColormap(cmap(cmapIndices),"cmapDiff")
+
+    for i in range(len(outputTimes)):
+        plot_global_comparison(
+            config,
+            lonTarg,
+            latTarg,
+            SST_MODEL[i,:,:],
+            SST_DATA[i,:,:],
+            SST_BIAS[i,:,:],
+            cmapModelObs,
+            clevsModelObs,
+            cmapDiff,
+            clevsDiff,
+            fileout = "%s/sstHADOI_%s_%s_years%04d-%04d.png" % (plots_dir, casename, outputTimes[i], climo_yr1, climo_yr2),
+            title = "SST (%s, years %04d-%04d)" % (outputTimes[i], climo_yr1, climo_yr2),
+            modelTitle = "%s" % casename,
+            obsTitle = "Observations (Hadley/OI, %s)" % preIndustrial_txt,
+            diffTitle = "Model-Observations",
+            cbarlabel = r"$^o$C")    
+            
+    
```

Comment:
Can you please provide a main type function to help modularize this and allow it to be run in script mode with command line arguments?  This helps generalize the code and increase its utility, e.g., something like

```python
if __name__ == "__main__":
    # process command line arguments and feed to main below
    main()
```